packages/dds/sequence/src/test/intervalRevertibles.fuzz.spec.ts

```diff
@@ -0,0 +1,376 @@
+		const start = startPosition(state);
+		const end = state.random.integer(start, Math.max(start, state.channel.getLength() - 1));
+		return { start, end };
+	}
+
+	function propertySet(state: ClientOpState): PropertySet {
+		const propNamesShuffled = [...options.propertyNamePool];
+		state.random.shuffle(propNamesShuffled);
+		const propsToChange = propNamesShuffled.slice(
+			0,
+			state.random.integer(1, propNamesShuffled.length),
+		);
+		const propSet: PropertySet = {};
+		for (const name of propsToChange) {
+			propSet[name] = state.random.string(5);
+		}
+		return propSet;
+	}
+
+	function nonEmptyIntervalCollection({ channel, random }: ClientOpState): string {
+		const nonEmptyLabels = Array.from(channel.getIntervalCollectionLabels()).filter((label) => {
+			const collection = channel.getIntervalCollection(label);
+			return isNonEmpty(collection);
+		});
+		return random.pick(nonEmptyLabels);
+	}
+
+	function interval(state: ClientOpState): { collectionName: string; id: string } {
+		const collectionName = nonEmptyIntervalCollection(state);
+		const intervals = Array.from(state.channel.getIntervalCollection(collectionName));
+		const id = state.random.pick(intervals)?.getIntervalId();
+		assert(id);
+
+		return {
+			id,
+			collectionName,
+		};
+	}
+
+	async function addText(state: ClientOpState): Promise<AddText> {
+		const { random, channel } = state;
+		return {
+			type: "addText",
+			index: random.integer(0, channel.getLength()),
+			content: random.string(random.integer(0, options.maxInsertLength)),
+		};
+	}
+
+	async function removeRange(state: ClientOpState): Promise<RemoveRange> {
+		return { type: "removeRange", ...exclusiveRange(state) };
+	}
+
+	async function addInterval(state: ClientOpState): Promise<AddInterval> {
+		return {
+			type: "addInterval",
+			...inclusiveRange(state),
+			collectionName: state.random.pick(options.intervalCollectionNamePool),
+			id: state.random.uuid4(),
+			// added to get rid of compiler errors - assume we
+			// don't want to test both features here
+			stickiness: 0,
+		};
+	}
+
+	async function deleteInterval(state: ClientOpState): Promise<DeleteInterval> {
+		return {
+			type: "deleteInterval",
+			...interval(state),
+		};
+	}
+
+	async function changeInterval(state: ClientOpState): Promise<ChangeInterval> {
+		const { start, end } = inclusiveRange(state);
+		return {
+			type: "changeInterval",
+			start: state.random.integer(0, 5) === 5 ? undefined : start,
+			end: state.random.integer(0, 5) === 5 ? undefined : end,
+			...interval(state),
+		};
+	}
+
+	async function changeProperties(state: ClientOpState): Promise<ChangeProperties> {
+		return {
+			type: "changeProperties",
+			...interval(state),
+			properties: propertySet(state),
+		};
+	}
+
+	async function revertSharedStringRevertibles(
+		state: ClientOpState,
+	): Promise<RevertSharedStringRevertibles> {
+		return {
+			type: "revertSharedStringRevertibles",
+			...interval(state),
```

Comment:
```suggestion
```

shouldn't be necessary to specify anything besides "do a revert"--revert is always the top of the stack.

(you'll need to make the "id" change I pointed out above for this to compile).

Also, once you do this, the generator doesn't actually need any randomness so if you're using `createWeightedGenerator`, you could just inline the `{ type: "revertSharedStringRevertibles" }` since it supports simple object shorthand for convenience when the object you're generating is a constant.